### PR TITLE
Adds -H:+GenerateBuildArtifactsFile, copies .so from remote container

### LINKED ISSUE #41020: Remote Container build does not copy all build artifacts
We are building a Quarkus native image in Gitlab Pipeline using Gradle and "remote container build" with Docker. 
Since we added https://quarkus.io/extensions/io.quarkiverse.poi/quarkus-poi/ extension and configured the Dockerfile according to Quarkus-POI guidelines (install and copy libraries), we faced java.awt.Font NoClassDefFoundError errors during generation of an Excel sheet.

> ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-7) HTTP Request to /reports/api/v1/download?start=2023-01&end=2024-12&source=ALL failed, error id: 6dc1b7cd-0365-4e01-b08c-b18f861228d9-1: java.lang.NoClassDefFoundError: Could not initialize class java.awt.Font

Debugging has shown the *.so files generated next to the *-runner application files are not copied back to the build output directory from the remote container. Building it locally by using just "container build" works as expected and the runtime error did not appear with the required *.so files next to the *-runner application file.

Debugging the build process confirmed: only the *-runner file is copied back from the container.
buildstep

Digging into the source we found there is no option to preserve the other produced build artifacts, just the executable will be copied back.

https://github.com/quarkusio/quarkus/blob/20d3c9d576d596ef3cdb90be7ce692afab63a7cb/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java#L57

It would be great to have all build artifacts preserved for further steps in the build process. 
For now we just copy the libs from a different location, but it would be great to just use them from remote container build.

> $ java --version
openjdk 17.0.10 2024-01-16
OpenJDK Runtime Environment (build 17.0.10+7-alpine-r0)
OpenJDK 64-Bit Server VM (build 17.0.10+7-alpine-r0, mixed mode, sharing)

> $ docker version
Client:
 Version:           20.10.22
 API version:       1.41
 Go version:        go1.18.9
 Git commit:        3a2c30b
 Built:             Thu Dec 15 22:21:58 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
Server: Docker Engine - Community
 Engine:
  Version:          20.10.22
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.9
  Git commit:       42c8b31
  Built:            Thu Dec 15 22:26:12 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.15
  GitCommit:        5b842e528e99d4d4c1686467debf2bd4b88ecd86
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

Build command: 
> ./gradlew build -Dquarkus.package.type=native -Dquarkus.native.remote-container-build=true -Dquarkus.native.native-image-xmx=8g -x test`

### PR DESCRIPTION
**Pull request by Karm**
fixes #41020 

This PR adds **to all native builds** the option `-H:+GenerateBuildArtifactsFile` that produces a small JSON file during the native-image build process. The JSON file contains a list of built artifacts.

We use that file as a manifest of what was built and what we need to copy from the remote container where the remote build took place. If no such file exists or if there are no extra artifacts to copy, we silently do nothing.

# What the new file looks like
```
~/workspaceRH/reproducers/poi$ cat ./build/poi-1.0.0-SNAPSHOT-native-image-source-jar/build-artifacts.json |jq
{
  "build_info": [
    "poi-1.0.0-SNAPSHOT-runner-build-output-stats.json"
  ],
  "executables": [
    "poi-1.0.0-SNAPSHOT-runner"
  ],
  "jdk_libraries": [
    "libawt.so",
    "libawt_headless.so",
    "libawt_xawt.so",
    "libfontmanager.so",
    "libfreetype.so",
    "libjavajpeg.so",
    "liblcms.so",
    "libmlib_image.so",
    "libjava.so",
    "libjvm.so"
  ]
}

```

# Why don't you podman cp?

Podman cp does not support globs, i.e. `cp /project/*.so` won't work, which I found surprising so I duly noted that in the comments in the code so as to save time for others who might be tempted to refactor it later.

### CODE REVIEW

**Comment by Karm** [comment]
Open for discussion, I think I could add a test, something like
```
integration-tests/awt-packaging/aws-lambda
integration-tests/awt-packaging/remote-container
```
to test that this PR keeps working and that this https://github.com/quarkusio/quarkus/pull/35718 doesn't break. It would create two additional builds, taking more time though. Perhaps a thing suited for quickstarts better. Not sure.

**Comment by geoand** [comment]
Do we need both those new integration tests? Can't we just have one?

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java [hunk 1]
```diff
@@ -34,59 +39,93 @@ protected void preBuild(Path outputDir, List<String> buildArgs) throws Interrupt
...
+                // The file is small enough to afford this read
+                final String buildArtifactsJson = Files.readString(buildArtifactsFile);
+                final Matcher m = JDK_LIBS.matcher(buildArtifactsJson);
+                if (m.matches()) {
+                    for (String lib : m.group(1).split(",")) {
+                        final String l = lib.trim().replace("\"", "");
+                        copyFromContainerVolume(outputDir, l,
+                                "Failed to copy " + l + " from container volume back to the host.");
+                    }
+                }
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java:91]: You may also use the JsonReader here, e.g.:

```suggestion
                final JsonObject jsonRead = JsonReader.of(buildArtifactsJson).read();
                JsonValue jdkLibraries = jsonRead.get("jdk_libraries");
                for (JsonValue lib : ((JsonArray) jdkLibraries).value()) {
                    copyFromContainerVolume(outputDir, ((JsonString) lib).value(),
                            "Failed to copy " + lib + " from container volume back to the host.");
                }
```

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java [hunk 2]
```diff
@@ -34,59 +39,93 @@ protected void preBuild(Path outputDir, List<String> buildArgs) throws Interrupt
...
         log.info(String.join(" ", command).replace("$", "\\$"));
-        Process process = new ProcessBuilder(command).start();
+        final Process process = new ProcessBuilder(command).start();
         if (process.waitFor() != 0) {
-            throw new RuntimeException(errorMsg);
+            throw new RuntimeException("Command failed: " + String.join(" ", command));
         }
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
-            return reader.readLine();
+            return reader.lines().toList();
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java:62]: What's the motivation behind this change? I understand that it's more generic this way but since we only use the method in a single place where this is not needed it seems to unnecessarily (for the time being) complicate things.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java [hunk 3]
```diff
@@ -34,59 +39,93 @@ protected void preBuild(Path outputDir, List<String> buildArgs) throws Interrupt
...
+
         if (nativeConfig.debug().enabled()) {
-            copyFromContainerVolume(outputDir, "sources", "Failed to copy sources from container volume back to the host.");
-            String symbols = String.format("%s.debug", nativeImageName);
-            copyFromContainerVolume(outputDir, symbols, "Failed to copy debug symbols from container volume back to the host.");
+            copyFromContainerVolume(outputDir, "sources",
+                    "Failed to copy sources from container volume back to the host.");
+            final String symbols = String.format("%s.debug", nativeImageName);
+            copyFromContainerVolume(outputDir, symbols,
+                    "Failed to copy debug symbols from container volume back to the host.");
```
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java:102]: Are these changes necessary?

**Review by zakkak — APPROVED** [review]
Thanks @Karm!

LGTM but it feels like touching more things than needed. I added some comments for your consideration.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java (hunk 2 above)
Comment (Karm) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java:62]: @zakkak I used that to read the output of `exec` for `find` to list the .so files in the started container. Then I abandoned that approach in favor of using the `-H:+GenerateBuildArtifactsFile`. So strictly speaking, we can leave in the thing that returns only the first line of output, however confusing that might be.

**Comment by Karm** [comment]
@geoand 
> Do we need both those new integration tests? Can't we just have one?

I wonder how to jam both into a single run. Perhaps we can. Like a project that has `quarkus.native.remote-container-build=true` in application.properties but the final artifact is `.zip` file "function" for AWS Lambda, right?  

Could be one :+1:

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java (hunk 3 above)
Comment (Karm) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java:102]: These changes are not necessary to fix the issue.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java (hunk 1 above)
Comment (Karm) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java:91]: @zakkak O.K., seemed to load way more stuff than needed for such a simple thing, but you are right. This is cleaner, it's JSON so let's read JSON...

**Review by gsmet — COMMENTED** [review]
I will let you adjust with Foivos comments but it's really cool that we can fix this.

This won't solve the case when you build your container using the Dockerfile though, right?

**Comment by Karm** [comment]
@geoand  

Commit 59b63bba0 adds a single new integration test that has in its application.properties:
```
quarkus.lambda.handler=test
quarkus.native.remote-container-build=true
```

It tests that both #41020 is fixed and that #35718 has no regression.

Tested as 
```
$ ./mvnw clean verify -f integration-tests -pl awt-packaging -Dnative -Dnative.surefire.skip=false
```

It passes, although **it produces these Lambda server related warnings**, I guess it could have something to do with the fact that the LambdaClient is deprecated...?

I tried to ask about it on Zulip:  https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Testing.20AWS.20Lambda.20locally/near/444730456 

```
[INFO] --- failsafe:3.2.5:integration-test (default) @ quarkus-integration-test-awt-packaging ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.it.jaxb.AwtJaxbTestIT
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
Executing "/home/karm/workspaceRH/quarkus/integration-tests/awt-packaging/target/quarkus-integration-test-awt-packaging-999-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/home/karm/workspaceRH/quarkus/integration-tests/awt-packaging/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus-internal.aws-lambda.test-api=localhost:5387"
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2024-06-14 18:42:08,760 INFO  [io.quarkus] (main) quarkus-integration-test-awt-packaging 999-SNAPSHOT native (powered by Quarkus 999-SNAPSHOT) started in 0.015s. Listening on: http://0.0.0.0:8081
2024-06-14 18:42:08,760 INFO  [io.quarkus] (main) Profile prod activated. 
2024-06-14 18:42:08,760 INFO  [io.quarkus] (main) Installed features: [amazon-lambda, awt, cdi, resteasy, resteasy-jaxb, smallrye-context-propagation, vertx]
2024-06-14 18:42:08,760 INFO  [io.qua.ama.lam.run.AbstractLambdaPollLoop] (Lambda Thread (NORMAL)) Listening on: http://localhost:5387/2018-06-01/runtime/invocation/next
2024-06-14 18:42:09,032 WARN  [io.qua.ama.lam.tes.LambdaClient] (main) LambdaClient has been deprecated and will be removed in future Quarkus versions.  You can now invoke using a built in test http server.  See docs for more details
2024-06-14 18:42:09,664 INFO  [io.qua.it.jax.Resource] (executor-thread-1) Received book: io.quarkus.it.jaxb.Book@2a3173ef
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.712 s -- in io.quarkus.it.jaxb.AwtJaxbTestIT
2024-06-14 18:42:12,661 ERROR [io.qua.ama.lam.run.AbstractLambdaPollLoop] (Lambda Thread (NORMAL)) Error running lambda (NORMAL): java.net.SocketException: Connection reset
        at java.base@21.0.3/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:318)
        at java.base@21.0.3/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
        at java.base@21.0.3/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
        at java.base@21.0.3/java.net.Socket$SocketInputStream.read(Socket.java:1099)
        at java.base@21.0.3/java.io.BufferedInputStream.fill(BufferedInputStream.java:291)
        at java.base@21.0.3/java.io.BufferedInputStream.read1(BufferedInputStream.java:347)
        at java.base@21.0.3/java.io.BufferedInputStream.implRead(BufferedInputStream.java:420)
        at java.base@21.0.3/java.io.BufferedInputStream.read(BufferedInputStream.java:399)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:827)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:759)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:952)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:759)
        at java.base@21.0.3/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1690)
        at java.base@21.0.3/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599)
        at java.base@21.0.3/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3235)
        at io.quarkus.amazon.lambda.runtime.AbstractLambdaPollLoop$1.run(AbstractLambdaPollLoop.java:95)
        at java.base@21.0.3/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21.0.3/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)

2024-06-14 18:42:12,661 ERROR [io.qua.run.StartupContext] (Lambda Thread (NORMAL)) Running a shutdown task failed [Error Occurred After Shutdown]: java.lang.RuntimeException: java.lang.InterruptedException
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$13.run(VertxHttpRecorder.java:888)
        at io.quarkus.runtime.StartupContext.runAllAndClear(StartupContext.java:87)
        at io.quarkus.runtime.StartupContext.close(StartupContext.java:79)
        at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
        at io.quarkus.runtime.Application.stop(Application.java:208)
        at io.quarkus.runtime.Application.stop(Application.java:155)
        at io.quarkus.amazon.lambda.runtime.AbstractLambdaPollLoop$1.run(AbstractLambdaPollLoop.java:165)
        at java.base@21.0.3/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21.0.3/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
Caused by: java.lang.InterruptedException
        at java.base@21.0.3/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1100)
        at java.base@21.0.3/java.util.concurrent.CountDownLatch.await(CountDownLatch.java:230)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$13.run(VertxHttpRecorder.java:886)
        ... 10 more

2024-06-14 18:42:12,662 INFO  [io.quarkus] (Lambda Thread (NORMAL)) quarkus-integration-test-awt-packaging stopped in 0.001s
2024-06-14 18:42:12,664 WARN  [io.net.boo.ServerBootstrap] (vert.x-acceptor-thread-0) Failed to register an accepted channel: [id: 0x94896071, L:/127.0.0.1:5387 ! R:/127.0.0.1:48170]: java.lang.IllegalStateException
        at io.vertx.core.net.impl.VertxEventLoopGroup.next(VertxEventLoopGroup.java:37)
        at io.vertx.core.net.impl.VertxEventLoopGroup.register(VertxEventLoopGroup.java:53)
        at io.netty.bootstrap.ServerBootstrap$ServerBootstrapAcceptor.channelRead(ServerBootstrap.java:241)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:97)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:553)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:840)

[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 

```

**Comment by Karm** [comment]
@zakkak 
I adjusted the changes, namely:
 - switching from regex to full blown Json reader
 
I left the changes regarding reading output there, both the error/debug ones and the runCommandAndReadOutput.
If you find it important that those are not touched, I find renaming those an alternative.

**Comment by Karm** [comment]
@gsmet 

> I will let you adjust with Foivos comments but it's really cool that we can fix this.
> 
Thx. I originally thought this would be 1 line of   :-D 
```
 containerRuntime.getExecutableName(), "cp",
                containerId + ":/project/*.so",
                outputDir.toAbsolutePath().toString()
```
but then 

```
Further note that podman cp does not support globbing (e.g., cp dir/*.txt). To copy multiple 
files from the host to the container use xargs(1) or find(1) (or similar tools for chaining commands) 
in conjunction with podman cp. To copy multiple files from the container to the host, use podman 
mount CONTAINER and operate on the returned mount point instead (see ALTERNATIVES below).
```
Source:  https://docs.podman.io/en/latest/markdown/podman-cp.1.html#description

> This won't solve the case when you build your container using the Dockerfile though, right?

This PR deals the scenario when you run your build "remotely" in a container, rather than just calling native-image from a container. The OP had a problem with Gradle, but it's not Gradle specific.

If you are building your **runtime** image using a Dockerfile, you still need to have a line in the Dockerfile that copies those files, e.g.   https://github.com/quarkusio/quarkus/blob/main/integration-tests/awt/src/main/docker/Dockerfile.native#L10  

Is there some kind of "Default" Dockerfile I can edit?

It used to be a problem because not everybody was using GraalVM/Mandrel 23.0+ and older versions did not produce additional .so files. Docker's COPY of non-existing files then triggered errors.  That required "hacks" like copying an unnecessary, but always present file, such as some .properties file, so as the COPY always copies something:
https://github.com/quarkusio/quarkus-quickstarts/blob/main/awt-graphics-rest-quickstart/src/main/docker/Dockerfile.native#L31

WDYT?

**Comment by gsmet** [comment]
> It used to be a problem because not everybody was using GraalVM/Mandrel 23.0+ and older versions did not produce additional .so files. Docker's COPY of non-existing files then triggered errors. That required "hacks" like copying an unnecessary, but always present file, such as some .properties file, so as the COPY always copies something

Yeah I think we should go with it if we are sure we always have a least one properties file. Be aware that we might not have an `application.properties` for instance. What was the file you counted on?

The Dockerfiles template for newly generated projects are in `tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles`.

**Comment by Karm** [comment]
@gsmet 
> > It used to be a problem because not everybody was using GraalVM/Mandrel 23.0+ and older versions did not produce additional .so files. Docker's COPY of non-existing files then triggered errors. That required "hacks" like copying an unnecessary, but always present file, such as some .properties file, so as the COPY always copies something
> 
> Yeah I think we should go with it if we are sure we always have a least one properties file. Be aware that we might not have an `application.properties` for instance. What was the file you counted on?
> 

It must have been one of the artifacts in target/, so I think `quarkus-artifact.properties`.
Is there a scenario where it wouldn't be present?

> The Dockerfiles template for newly generated projects are in `tools/base-codestarts/src/main/resources/codestarts/quarkus/tooling/dockerfiles`.

Gonna take a look.

**Comment by Karm** [comment]
Ad "getting verticle", a message per my workstation core, why is it printing there? It's gonna be 256 messages on dualsocket Altra Max :) 

```
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.it.jaxb.AwtJaxbTestIT
getting verticle
...
getting verticle
```

...and indeed, I've juts tried on an 80 core machine and there is **80** `getting verticle` messages.

**Comment by geoand** [comment]
@Karm I guess you overcame the native image test you were having :)

**Review by zakkak — APPROVED** [review]
LGMT

**Comment by Karm** [comment]
> @Karm I guess you overcame the native image test you were having :)

Well, I wish I had. I used the deprecated API LambdaClient, which feels suboptimal for a new test...

**Comment by geoand** [comment]
Sure yeah, but we can figure it out later

**Comment by Karm** [comment]
Hello, are there any action items for me?
